crates/attestation/src/crypto/crl.rs

```diff
@@ -57,18 +49,18 @@ impl<'a> PkiPathCRLCheck<'a> for PkiPath<'a> {
                 break;
             }
             let cert = cert.unwrap();
-            let next = next.unwrap();
+            let next = *next.unwrap();
 
             let crls = {
                 let mut crls = vec![];
-                if let Some(crl) = pairs.get_crl_by_issuer(&cert.tbs_certificate.subject) {
-                    crls.push(crl);
-                } else {
-                    let urls = cert.tbs_certificate.get_crl_urls()?;
-                    for url in urls {
-                        if let Some(crl) = pairs.get_crl_by_url(&url) {
-                            crls.push(crl);
-                        }
+                let mut urls = next.tbs_certificate.get_crl_urls()?;
+                // Some end certs use the parent cert CRL
```

Comment:
This should never be the case I believe, could you give me the example of where this happened?

Reply:
The AMD CPU certs don't have CRLs

Reply:
If they don't have a CRL listed, that would mean they can't be revoked individually, but only at the level up.
Looking at a different CRL implicitly can be highly problematic:
If you have a chain Root -> Intermediate -> Leaf, where Leaf happens to get the same serial as a revoked (unrelated) Intermediate, we'd also consider the Leaf revoked, even though it's not supposed to be.